absl/strings/escaping_test.cc

```diff
@@ -635,4 +635,99 @@ TEST(HexAndBack, HexStringToBytes_and_BytesToHexString) {
   EXPECT_EQ(hex_only_lower, hex_result);
 }
 
+TEST(NonNullTerminatedStrings, OctalConversion) {
+  absl::string_view reference("\\111", 2);
+  std::string expected = "\1";
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, expected);
+}
+
+TEST(NonNullTerminatedStrings, InvalidHexConversion) {
+  absl::string_view reference("\\x1", 2);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(reference, &unescaped_str));
+}
+
+TEST(NonNullTerminatedStrings, HexConversion) {
+  absl::string_view reference("\\x11", 3);
+  std::string expected = "\x1";
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+  EXPECT_EQ(unescaped_str, expected);
+}
+
+TEST(NonNullTerminatedStrings, UtfConversion_u) {
+  absl::string_view invalid("\\u11111", 5);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(invalid, &unescaped_str));
+}
+
+TEST(NonNullTerminatedStrings, UtfConversion_U) {
+  absl::string_view invalid("\\U0000000000000", 9);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(invalid, &unescaped_str));
+}
+
+TEST(NonNullTerminatedStrings, InvalidOctal) {
+  absl::string_view reference("\\1z1", 3);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\1z");
+}
+
+TEST(NonNullTerminatedStrings, InvalidHex) {
+  absl::string_view reference("\\xz", 3);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(reference, &unescaped_str));
+}
+
+TEST(NonNullTerminatedStrings, InvalidUtf_u_RightSizeButInvalidChar) {
```

Comment:
Google Test asks that the names of tests not contain underscores: https://github.com/google/googletest/blob/master/googletest/docs/FAQ.md#why-should-not-test-case-names-and-test-names-contain-underscore

Reply:
Ouch, rookie mistake. Thanks!

Reply:
It's hard to make that look good without underscores :(

I also noticed some other tests have underscores. That could be a low hanging issue for new comers.

Reply:
Definitely not a rookie mistake, as evidenced by the other code that violates it :)

Sure, this could be a fine thing for someone wanting to get their feet wet.